.github/workflows/publish.yml

```
@@ -1,23 +1,38 @@
-name: Kirby release CI
+name: Publish to npm
 
 on:
-  push:
-    tags:
-      - '^v[0-9]+\.[0-9]+'
+  pull_request:
+    types: closed
+    
 defaults:
   run:
     shell: bash
 
 jobs:
-  publish_to_npm:
+  publish:
     runs-on: ubuntu-latest
+    if: github.event.pull_request.merged
     steps:
       - uses: actions/checkout@v2
+        with:
+          ref: refs/heads/master      
       - name: Install Node.js
         uses: actions/setup-node@v2
         with:
           node-version: '10.16'
-      - name: Create npm config
-        run: echo '//registry.npmjs.org/:_authToken=${{secrets.NPM_PUBLISH_TOKEN}}' > ~/.npmrc
+      - name: Install npm
+        run: npm install
+      - name: Configure git
+        run: |
+          git config user.name "Kirby Bot"
+          git config user.email "<>"
+      - name: Create .npmrc
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_PUBLISH_TOKEN }}
+        run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
```

Simplify (revert/keep existing script)
```suggestion
        run: echo '//registry.npmjs.org/:_authToken=${{secrets.NPM_PUBLISH_TOKEN}}' > ~/.npmrc
```